fsspec/core.py

```diff
@@ -334,7 +334,16 @@ def _un_chain(path, kwargs):
         kws = kwargs.pop(protocol, {})
         if bit is bits[0]:
             kws.update(kwargs)
-        kw = dict(**extra_kwargs, **kws)
+        if any(
```

Comment:
The simpler alternative would be
```
kw = extra_kwargs | kws
```
alone (you have this some lines below) without the check.

which does something different, but maybe the right thing?